src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4035,10 +4037,18 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers)
                 {
                     case SyntaxKind.ThisKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureExtensionMethod);
+                        if (seenRefModifier)
+                        {
+                            modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefExtensionMethods);
+                        }
                         break;
                     case SyntaxKind.InKeyword:
                     case SyntaxKind.ReadOnlyKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureReadonlyReferences);
+                        seenRefModifier = true;
```

Comment:
yes, only for "in this", "ref this" and "ref readonly this"

---
In reply to: [123672918](https://github.com/dotnet/roslyn/pull/20403#discussion_r123672918) [](ancestors = 123672918,123672189)